nautobot/core/models/__init__.py

```diff
@@ -34,6 +36,29 @@ def present_in_database(self):
         """
         return not self._state.adding
 
+    def get_absolute_url(self):
+        """
+        Return the canonical URL for this object.
+        """
+
+        # Iterate the pk-like fields and try to get a URL, or return None.
+        fields = ["pk", "slug"]  # TODO: Eventually all PKs
+        actions = ["retrieve", "detail", ""]  # TODO: Eventually all retrieve
+
+        for field in fields:
+            if not hasattr(self, field):
+                continue
+
+            for action in actions:
+                route = get_route_for_model(self, action)
+
+                try:
+                    return reverse(route, kwargs={field: getattr(self, field)})
+                except NoReverseMatch:
+                    continue
+
+        return None
```

Comment:
Wonder if it might be better from a Django API standpoint to raise `AttributeError` (as if the model didn't have a `get_absolute_url` function at all) in the failure case rather than returning `None`? I know it may make for needing more exception handling in our code but it might be worth it.